### PR TITLE
Added custom notifications

### PR DESCRIPTION
**Pull request by Aga-C**
I've moved code from my old pull request from SMT adding custom notifications to the app. 

Short summary of changes:
- Based on Android notification channels. For Androids older than 8.0 the option is invisible.
- Notification channel ID is a conversation ID, so it's proof for name changes. Channel name is a conversation name.
- There's an ability to remove custom notification from the conversation, what's not a typical option in SMS apps (but I think it's very useful).

It addresses all requests from original Simple SMS Messenger related to changing sound, silence conversations, overriding do-not-disturb etc., because all of these settings are available in system settings for notification channel. They can differ between various distributions of Android, but what's here can be the basis for further enhancements in case some devices don't support specific options. 

https://github.com/FossifyOrg/SMS-Messenger/assets/85929121/f7d25bf4-9a8c-4c91-bafe-58ab3cbc50d1

### CODE REVIEW

**Comment by naveensingh** [comment]
@Aga-C I'll check this later. 

Maybe it should "Enable custom notifications" in place of "Use custom notifications"?

**Comment by naveensingh** [comment]
@Aga-C 

There is no ripple when I click on either of the options. Please use the same style as in preferences. Check `activity_settings.xml` for reference. 

That is, use `style="@style/SettingsHolderCheckboxStyle"` and `style="@style/SettingsHolderTextViewOneLinerStyle"` and remove the paddings/margins.

**Comment by naveensingh** [comment]
Actually, the other options there (headings, conversation name, members) won't be aligned properly after applying the preference style so if you want, you can fix them as well using the appropriate style.

**Comment by Aga-C** [comment]
I've changed it, but I encountered two problems:
- There is no ripple on a button, but maybe it has something to do with opening system settings?
- I couldn't set proper styles for Members section. RecyclerView seems to enforce some margins.

**Comment by naveensingh** [comment]
@Aga-C ok, I'll fix those.

File: app/src/main/kotlin/org/fossify/smsmessenger/activities/ConversationDetailsActivity.kt [hunk 1]
```diff
@@ -60,6 +74,62 @@ class ConversationDetailsActivity : SimpleActivity() {
...
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun createNotificationChannel() {
+        val name = conversation?.title
+        val audioAttributes = AudioAttributes.Builder()
+            .setUsage(AudioAttributes.USAGE_NOTIFICATION)
+            .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+            .setLegacyStreamType(AudioManager.STREAM_NOTIFICATION)
+            .build()
+
+        NotificationChannel(threadId.hashCode().toString(), name, NotificationManager.IMPORTANCE_HIGH).apply {
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/smsmessenger/activities/ConversationDetailsActivity.kt:119]: @Aga-C Is there any reason we are using `threadId.hashCode().toString()` and not `threadId.toString()`?

Comment (naveensingh) [reply app/src/main/kotlin/org/fossify/smsmessenger/activities/ConversationDetailsActivity.kt]: Changed it to just `threadId.toString()` as mapping to Int is not required in this case. The notification IDs is required to be an Int.

**Comment by naveensingh** [comment]
Tested it on a couple of devices, seems to work fine. Thanks :)

**Pull request merged**